src/trio/_core/_parking_lot.py

```diff
@@ -234,6 +266,23 @@ def repark_all(self, new_lot: ParkingLot) -> None:
         """
         return self.repark(new_lot, count=len(self))
 
+    def break_lot(self, task: Task) -> None:
+        """Break this lot, causing all parked tasks to raise an error, and any
+        future tasks attempting to park (and unpark? repark?) to error. The error
+        contains a reference to the task sent as a parameter."""
+        self.broken_by = task
+        # TODO: weird to phrase this one, we probably should reraise this error in Lock
+        error = outcome.Error(
+            _core.BrokenResourceError(f"Parking lot broken by {task}"),
+        )
```

Comment:
I think this shouldn't be cached, each task should get its own outcome & exception object. Otherwise the traceback would get scrambled together.

Reply:
thanks, fixed